pkg/platform/api/buildplanner/response/impactreport.go

```diff
@@ -34,10 +49,26 @@ type ImpactReportError struct {
 
 func (e ImpactReportError) Error() string { return e.Message }
 
+func IsImpactReportBuildPlanningError(err error) bool {
+	var impactReportErr *ImpactReportError
+	return errors.As(err, &impactReportErr) && impactReportErr.Type == buildPlanningType
+}
+
 func ProcessImpactReportError(err *ImpactReportResult, fallbackMessage string) error {
 	if err.Error == nil {
 		return errs.New(fallbackMessage)
 	}
 
-	return &ImpactReportError{err.Type, err.Message}
+	errType := err.Type
+	message := err.Message
+	buildResultError := err.ImpactReportBuildResultError
+	if beforeType := buildResultError.BuildBefore.Type; beforeType != buildPlannedType {
+		errType = beforeType
+		message += "\nbuildBefore status: " + beforeType
```

Comment:
Can you comment why we care about the buildBefore? I'm not sure what this is telling us exactly, but based on what I can grok from here it sounds like this is telling us what the buildstatus was before we evaluating (ie. triggered) it. Why do we care about this?

Perhaps the impactReport API doesn't trigger builds, and so this is telling us that it's not planned, which effectively means it won't be planned unless some other external call happens. In which case retrying won't make any difference.

I think either we need to document the logic or we need to revise the logic.

Reply:
The error message we were getting was:

```
Could not resolve one or both input buildplans
```

Therefore I think we need to check the status of both input build expressions. If we only check afterExpr, we could open ourselves up to a subtle bug.

Recently I had an issue with the fact that the Platform did not recognize that a local commit belonged to a project (because it wasn't pushed yet). I can envision a scenario where a local commit's build expression submitted as `beforeExpr` could fail in a similar way.

Reply:
Is it possible to make two commits in rapid succession such that the second commit is using the first commit as `beforeExpr`, and that `beforeExpr` may not have resolved yet in time for the second commit's call to ImpactReport? I agree that ImpactReport is not doing any triggering, but we are (StageCommit, etc.). It's just that I don't think we cannot guarantee that the trigger for `beforeExpr` is fully resolved every time.

Also, I've had experience with local commits not counted as belonging to the current project, so I can envision a scenario where a local commit's build expression can be the `beforeExpr` and fail to resolve for a while.